statistics/table.go

```diff
@@ -465,18 +465,35 @@ func outOfRangeEQSelectivity(ndv, modifyRows, totalRows int64) float64 {
 }
 
 // getEqualCondSelectivity gets the selectivity of the equal conditions.
-func (coll *HistColl) getEqualCondSelectivity(idx *Index, bytes []byte, usedColsLen int) float64 {
+func (coll *HistColl) getEqualCondSelectivity(sc *stmtctx.StatementContext, idx *Index, bytes []byte, usedColsLen int, colRanges []*ranger.Range) (float64, error) {
+	minRowCount := math.MaxFloat64
+	cols := coll.Idx2ColumnIDs[idx.ID]
+	for i, colID := range cols {
+		if i >= usedColsLen {
+			break
+		}
+		if col, ok := coll.Columns[colID]; ok {
+			rowCount, err := col.GetColumnRowCount(sc, colRanges, coll.ModifyCount, col.IsHandle)
```

Comment:
The `colRanges` may contain point ranges of other columns. For example, `a = 1 and b = 2 and c = 3`, `GetColumnRowCount` would use `[1,1]` for all these 3 columns, which is wrong.

Reply:
You can extract the point range related to this column from the argument `colRanges` here. @tangwz